lib/rules/array-callback-return.js

```diff
@@ -79,7 +76,14 @@ function getArrayMethodName(node) {
             case "ReturnStatement": {
                 const func = astUtils.getUpperFunction(parent);
 
-                if (func === null || !astUtils.isCallee(func)) {
+                if (
+                    func === null ||
+                    !astUtils.isCallee(
+                        func.parent.type === "ChainExpression"
```

Comment:
I think this can never happen? 

The existing `func === null || !astUtils.isCallee(func)` condition seems fine, `case "ChainExpression"` will handle `func.parent.parent` if needed.